src/core/Directus/Database/Schema/SchemaFactory.php

```diff
@@ -154,6 +160,19 @@ public function alterTable($name, array $data)
         }
 
         $toChangeColumnsData = ArrayUtils::get($data, 'change', []);
+
+        // Throws an exception when trying to make the field required and there are items with no value for that field in collection
+        foreach ($toChangeColumnsData as $column) {
+            if($column['required']) {
+                $selectQuery = sprintf('SELECT * FROM `%s` WHERE `%s` IS NULL',$name,$column['field']);
```

Comment:
Let's change this to `SELECT COUNT` instead, otherwise it would fetch the whole table just to do this check.